#HW3 Assignment 2
#August 2015
#Author: NDW (reference from FBB)

In [111]:
#import os
#import sys
import numpy as np
#import pylab as pl


#NULL HYPOTHESIS: the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05¶
 

In [112]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=.272 
P_1=.282

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

In [113]:
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )

sp_stdev_3y=sp_stdev((P_0+P_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_3y 

0.272 409 564 0.0322832705591


In [114]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(P_1, P_0, sp_stdev_3y)
print z_3y

0.309757959054


In [115]:
p_3y=1-0.6217

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_3y,alpha)

is the p value 0.38 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


#NULL HYPOTHESIS: the % of former prisoners employed for 6 or more quarters after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05¶
 

In [116]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


#since 1.589 is less than 3.84, and our p-value is > 0.05, we fail to reject the null hypothesis.